apps/obsidian/src/components/TldrawViewComponent.tsx

```diff
@@ -21,9 +38,84 @@ export const TldrawPreviewComponent = ({
     return () => clearTimeout(timer);
   }, []);
 
+  const saveChanges = useCallback(async () => {
+    const newData = getTLDataTemplate({
+      pluginVersion: plugin.manifest.version,
+      tldrawFile: createRawTldrawFile(store),
+      uuid: window.crypto.randomUUID(),
+    });
+    const stringifiedData = JSON.stringify(newData, null, "\t");
+
+    if (stringifiedData === lastSavedDataRef.current) {
+      return;
+    }
+
+    const currentContent = await plugin.app.vault.read(file);
+    if (!currentContent) {
+      console.error("Could not read file content");
+      return;
+    }
+
+    const updatedString = getUpdatedString(currentContent, stringifiedData);
+    if (updatedString === currentContent) {
+      return;
+    }
+
+    try {
+      await plugin.app.vault.modify(file, updatedString);
+
+      const verifyContent = await plugin.app.vault.read(file);
+      const verifyMatch = verifyContent.match(
+        new RegExp(
+          `${TLDATA_DELIMITER_START}\\s*([\\s\\S]*?)\\s*${TLDATA_DELIMITER_END}`,
+        ),
+      );
+
+      if (!verifyMatch || verifyMatch[1]?.trim() !== stringifiedData.trim()) {
+        throw new Error("Failed to verify saved TLDraw data");
+      }
+
+      lastSavedDataRef.current = stringifiedData;
+    } catch (error) {
+      console.error("Error saving/verifying TLDraw data:", error);
+      // Reload the editor state from file since save failed
+      const fileContent = await plugin.app.vault.read(file);
+      const match = fileContent.match(
+        new RegExp(
+          `${TLDATA_DELIMITER_START}([\\s\\S]*?)${TLDATA_DELIMITER_END}`,
+        ),
+      );
+      if (match?.[1]) {
+        const data = JSON.parse(match[1]);
+        if (data.raw) {
+          loadSnapshot(store, data.raw);
+        }
+      }
+    }
+  }, [file, plugin, store]);
+
+  useEffect(() => {
+    const unsubscribe = store.listen(
+      () => {
+        if (saveTimeoutRef.current) {
+          clearTimeout(saveTimeoutRef.current);
+        }
+        saveTimeoutRef.current = setTimeout(saveChanges, DEFAULT_SAVE_DELAY);
+      },
+      { source: "user", scope: "document" },
+    );
+
+    return () => {
+      unsubscribe();
+      if (saveTimeoutRef.current) {
+        clearTimeout(saveTimeoutRef.current);
+      }
+    };
+  }, [store, saveChanges]);
+
   const handleMount = useCallback((editor: Editor) => {
-    editor.setCurrentTool("hand");
-    editor.updateInstanceState({});
+    editorRef.current = editor;
+    editor.setCurrentTool("select");
 
     const shapes = editor.getCurrentPageShapes();
     if (shapes.length > 0) {
```

Comment:
We should aim to eventually save the last view so that when the canvas is loaded again, it is in the same place. This can be deferred to later.  If `zoomToFit` is requested by users, we can add that as a config option later.